modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -6054,6 +6054,17 @@ bool GDScriptAnalyzer::check_type_compatibility(const GDScriptParser::DataType &
 				valid = p_target.get_container_element_type(0) == p_source.get_container_element_type(0);
 			}
 		}
+		if (valid && p_target.builtin_type != Variant::ARRAY && p_source.builtin_type == Variant::ARRAY) {
```

Comment:
This only checks for `p_target.builtin_type != Variant::ARRAY` so anything that isn't an Array. That would include all other Variant types including Dictionary, float, etc., no?

Reply:
It's a lil confusing, but the implementation of the `Variant::can_convert_strict` call above here (line 6025) guarantees that if `valid == true`, and  `p_source is an ARRAY`, then `p_target is either an ARRAY or a PACKED_ARRAY`. I can try to make this more clear, since this could be a bug farm if the implementation of `Variant::can_convert_strict` changes.